pkg/generators/repo_host.go

```diff
@@ -0,0 +1,114 @@
+package generators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/services/repo_host"
+)
+
+var _ Generator = (*RepoHostGenerator)(nil)
+
+const (
+	DefaultRepoHostRequeueAfterSeconds = 30 * time.Minute
+)
+
+type RepoHostGenerator struct {
+	client client.Client
+	// Testing hooks.
+	overrideHost repo_host.RepoHostService
+}
+
+func NewRepoHostGenerator(client client.Client) Generator {
+	return &RepoHostGenerator{client: client}
+}
+
+func (g *RepoHostGenerator) GetRequeueAfter(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) time.Duration {
+	// Return a requeue default of 30 minutes, if no default is specified.
+
+	if appSetGenerator.RepoHost.RequeueAfterSeconds != nil {
+		return time.Duration(*appSetGenerator.RepoHost.RequeueAfterSeconds) * time.Second
+	}
+
+	return DefaultRepoHostRequeueAfterSeconds
+}
+
+func (g *RepoHostGenerator) GetTemplate(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) *argoprojiov1alpha1.ApplicationSetTemplate {
+	return &appSetGenerator.RepoHost.Template
+}
+
+func (g *RepoHostGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator, applicationSetInfo *argoprojiov1alpha1.ApplicationSet) ([]map[string]string, error) {
```

Comment:
I think it would b better to pass only the metadata, so the function get's only the relevant part and the whole spec

Reply:
Passing the whole object does leave a lot of nice doors open though, especially having generators that set Status conditions or having more global config in the future.